modules/os2web_audit_entity/src/Form/SettingsForm.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Drupal\os2web_audit_entity\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class SettingsForm.
+ *
+ * This is the settings for the module.
+ */
+class SettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    ConfigFactoryInterface $configFactory,
+    private EntityTypeManagerInterface $entityTypeManager,
+  ) {
+    parent::__construct($configFactory);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): static {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * The name of the configuration setting.
+   *
+   * @var string
+   */
+  public static string $configName = 'os2web_audit_entity.settings';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames(): array {
+    return [self::$configName];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'os2web_audit_entity_admin_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $items = [];
+    $roles = $this->getRoles();
+    foreach ($roles as $role) {
+      $items[$role->id()] = $role->label();
+    }
+
+    $config = $this->config(self::$configName);
+
+    $form['roles'] = [
+      '#type' => 'checkboxes',
+      '#title' => $this->t('Choose an Option'),
+      '#description' => $this->t('Please select an option from the dropdown menu.'),
+      '#options' => $items,
+      '#default_value' => $config->get('roles') ?? [],
+      '#required' => TRUE,
+    ];
```

Comment:
We should probably say that selected roles are considered API-roles and what that means with respect to logging.

Reply:
Fixed